Python_Engine/Compute/Run.cs

```diff
@@ -0,0 +1,225 @@
+
+                process.StartInfo = startInfo;
+                process.Start();
+            });
+        }
+
+        [Description("Run a command via CMD and return True if successful and False if not.")]
+        [Input("command", "The command to be run.")]
+        [Input("hideWindows", "Set to True to hide cmd windows.")]
+        [Input("startDirectory", "The directory in which the command should be run.")]
+        [Output("success", "True if successful and False if not.")]
+        public static bool RunCommandBool(string command, bool hideWindows = false, string startDirectory = null)
+        {
+            System.Diagnostics.Process process = new System.Diagnostics.Process();
+
+            string commandMode = "/K";
+            if (hideWindows)
+            {
+                process.StartInfo.CreateNoWindow = true;
+                process.StartInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+                commandMode = "/C";
+            }
+            process.StartInfo.FileName = "cmd.exe";
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.Arguments = $"{commandMode} {command}";
+            process.Start();
+
+            process.WaitForExit();
+
+            if (process.ExitCode != 0)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        [Description("Run a command via CMD and return stdout.")]
+        [Input("command", "The command to be run.")]
+        [Input("hideWindows", "Set to True to hide cmd windows.")]
+        [Input("startDirectory", "The directory in which the command should be run.")]
+        [Output("stdout", "The StandardOutput from the command that was run. If the process failed, then StandardError will be returned here instead.")]
+        public static string RunCommandStdout(string command, bool hideWindows = true, string startDirectory = null)
+        {
+            System.Diagnostics.Process process = new System.Diagnostics.Process();
+
+            string commandMode = "/K";
+            if (hideWindows)
+            {
+                process.StartInfo.CreateNoWindow = true;
+                process.StartInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+                commandMode = "/C";
+            }
+            process.StartInfo.FileName = "cmd.exe";
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.RedirectStandardError = true;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.Arguments = $"{commandMode} {command}";
+            process.Start();
+
+            // To avoid deadlocks, always read the output stream first and then wait.  
+            string stdout = process.StandardOutput.ReadToEnd();
+            string stderr = process.StandardError.ReadToEnd();
+            process.WaitForExit();
```

Comment:
What happens if the process hangs indefinitely without exiting? I think this way the process may deadlock Grasshopper's UI. Shouldn't we consider something like:
```suggestion
            int timeoutMinutes = 5; // should be an input parameter to the method
            int millisecondsToWait = 5 * 60 * 1000;
            process.WaitForExit(millisecondsToWait);
            if (!myProcess.HasExited)
		process.Kill();
```